[FEAT] Show crop details on hover (seedling)

tiffanydys opened the pull request:
# Description
- provides information on crop seedling upon hover

Fixes #issue suggestion by @ Reymar in discord chat

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
`yarn test`
<img width="402" alt="Screen Shot 2022-03-28 at 3 26 35 AM" src="https://user-images.githubusercontent.com/18549210/160297446-9404ad80-f53a-4636-b1ae-f6cd62828c65.png">

`yarn dev`
Speed planting scenario:

https://user-images.githubusercontent.com/18549210/160297265-0c888b34-fd85-47cb-8766-61452648475c.mov



# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]
- [x] Screenshot if it includes any UI changes
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

adamhannigan commented:
I like this idea. Lots of people get confused around some of the seedlings (i.e. potato and carrot). 😅 

2 small suggestions:

1. Show the tooltip when hovering anywhere over the plant (it only shows when hovering over the time bar)
2. Show the plant name + the crop icon. The time seems redundant since they can already see that.

tiffanydys commented:
> I like this idea. Lots of people get confused around some of the seedlings (i.e. potato and carrot). 😅
> 
> 2 small suggestions:
> 
> 1. Show the tooltip when hovering anywhere over the plant (it only shows when hovering over the time bar)
> 2. Show the plant name + the crop icon. The time seems redundant since they can already see that.

Hi adam, added the crop icon and removed time left details, also updated the video for your verification 😄. Thanks!

adamhannigan commented:
UI looks more appealing now @tiffanydys 👏 well done

I've found that it gets a bit annoying when planting very quickly. I.e. planting 20 crops, it just gets in the way.

Is it possible not to show the tooltip when the user just planted the crop?

Z1kou commented:
> UI looks more appealing now @tiffanydys 👏 well done
> 
> I've found that it gets a bit annoying when planting very quickly. I.e. planting 20 crops, it just gets in the way.
> 
> Is it possible not to show the tooltip when the user just planted the crop?

maybe alittle delay of the tooltip show up time would be lovely , for a few seconds perhaps ! Also @tiffanydys  amazing job !

tiffanydys commented:
> UI looks more appealing now @tiffanydys 👏 well done
> 
> I've found that it gets a bit annoying when planting very quickly. I.e. planting 20 crops, it just gets in the way.
> 
> Is it possible not to show the tooltip when the user just planted the crop?

Hi adam, made it possible with `setTimeout` function, took a while but it finally works 😄
Thanks to @Z1kou for giving me the delayed tooltip idea 💡

adamhannigan commented on src/features/crops/components/Field.tsx:
```diff
@@ -42,7 +44,33 @@ export const Field: React.FC<Props> = ({
 
+  let timeoutId: null | ReturnType<typeof setTimeout> = null
+
+  const handleMouseHover = () => {
```
We may be able to take a simpler approach here using the field data. In this case could we check that the field was recently planted. @tiffanydys 

I.e.

```
const handleMouseHover = () => {
    // Field was just planted
    if (field.plantedAt < Date.now() - 1000) {
        return;
    }
    setShowCropDetails(true);
}
```
Setting up listeners can get a bit dangerous like this as we need to manage them and ensure we clean them up in the case the component unmounts.

tiffanydys replied:
Hi @adamhannigan,
I think you meant `now - field.plantedAt < 1000`
I've followed your approach on my recent update, it's much cleaner now and easier to understand 😄 

I've updated the code and the video evidence. Please test it out when you got time. ✌🏻
Thanks for your guidance ❤️😄

adamhannigan reviewed (approved):
Great work as always @tiffanydys 👏
